ImageGoNord/GoNord.py

```diff
@@ -417,7 +425,70 @@ def converted_loop(self, is_rgba, pixels, original_pixels, maxRow, maxCol, minRo
                 pixels[row, col] = tuple(colors_list)
         return pixels
 
-    def convert_image(self, image, save_path='', parallel_threading=False):
+    def convert_image_by_model(self, image, use_model_cpu=False):
+        """
+        Process a Pillow image by using a PyTorch model "PaletteNet" for recoloring the image
+
+        Parameters
+        ----------
+        image : pillow image
+            The source pillow image
+        use_model_cpu : bool, optional
+            true if using cpu power
+
+        Returns
+        -------
+        pillow image
+            processed image
+        """
+        FE = FeatureEncoder() # torch.Size([64, 3, 3, 3])
+        RD = RecoloringDecoder() # torch.Size([530, 256, 3, 3])
+
+        FE.load_state_dict(torch.load(pkg_resources.open_binary(palette_net, "FE.state_dict.pt")))
+        RD.load_state_dict(torch.load(pkg_resources.open_binary(palette_net, "RD.state_dict.pt")))
+
+        if use_model_cpu:
+            FE.to("cpu")
+            RD.to("cpu")
+
+        lab_image = ((convertor.rgb2lab(np.array(image))) - [50,0,0] ) / [50,127,127]
+
+        img = torch.Tensor(lab_image).permute(2,0,1)
+
+        h = 16*int(img.shape[1]/16)
+        w = 16*int(img.shape[2]/16)
+
+        T = transforms.Resize((h,w))
+
+        img = T(img)
+        img = img.unsqueeze(0)
+        palette = []
+        for hex, rgb_value in self.PALETTE_DATA.items():
+            a = []
+            for j in [2,4,6]:
+                a.append(int(hex[j-2:j],16))
+            palette.append(a)
+
+        try:
+            pal_np = np.array(palette).reshape(1,6,3)/255
+        except:
+            print("You have too many colors in your palette for the model. Auto reducing the array. It's limited at 6 colors, now you have: ", len(palette))
```

Comment:
I dont get the message inside the print, do you want to comunicate that the only color that will be taken will be the first 6? It is clear in line 476, but can we be a more explicit?

```suggestion
            print("You have too many colors in your palette for the model, this feature is limited to 6 colours, now you have: ", len(palette), "! I'll take the first 6!")
```

Or something like this.

Maybe we can also throw an error without proceed further and just print the reason in the error.